cmd/zc_traverser_blob.go

```diff
@@ -162,18 +164,33 @@ func (t *blobTraverser) Traverse(preprocessor objectMorpher, processor objectPro
 	blobProperties, isBlob, isDirStub, propErr := t.getPropertiesIfSingleBlob()
 
 	if stgErr, ok := propErr.(azblob.StorageError); ok {
+
 		// Don't error out unless it's a CPK error just yet
 		// If it's a CPK error, we know it's a single blob and that we can't get the properties on it anyway.
 		if stgErr.ServiceCode() == common.CPK_ERROR_SERVICE_CODE {
-			return errors.New("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. " +
-				"If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs")
+			if t.getProperties {
+				return errors.New("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. " +
+					"If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs")
+			} else {
+				blobProperties = &azblob.BlobGetPropertiesResponse{}
+				isBlob = true
```

Comment:
Both being true usually conflicts enumerator's decisions. Have we checked the impact? Otherwise, we could pass rename the getProperties flag to isRemoveEnumerator and check only for listing.

Reply:
Why would it conflict with the enumerators decisions? What would azcopy do if you had a marker file where the directory exists